draft-ietf-protected-audience-key-value-services.md

```diff
@@ -0,0 +1,520 @@
+following schema for the client to
+successfully parse them.
+
+Note that they must be serialized to string when stored as the value.
+
+#### InterestGroupResponse
+
+The schema below is defined following the spec by
+https://json-schema.org. For values for keys
+from the `interestGroupNames` namespace, they must conform to the
+following schema, prior to being
+serialized to string.
+
+~~~~~ json
+{
+    "title": "tkv.response.v2.InterestGroupResponse",
+    "description": "Format for value of keys in groups tagged 'interestGroupNames'",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "priorityVector": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "description": "signals",
+                    "type": "number"
+                }
+            }
+        },
+        "updateIfOlderThanMs": {
+            "description": "This optional field specifies that the interest group should be updated if the interest group hasn't been joined or updated in a duration of time exceeding `updateIfOlderThanMs` milliseconds. Updates that ended in failure, either parse or network failure, are not considered to increment the last update or join time. An `updateIfOlderThanMs` that's less than 10 minutes will be clamped to 10 minutes.",
+            "type": "unsigned integer"
+        }
+    }
+}
+~~~~~
+
+Example:
+
+~~~~~ json
+{
+    "priorityVector": {
+        "signal1": 1,
+        "signal2": 2
+    },
+    "updateIfOlderThanMs": 10000
+}
+~~~~~
+
+### Generating a Response
+
+This algorithm describes how the Key Value Service MAY generate a response to a request.
+
+The input is a `payload` corresponding to {{response-schema}} and the HPKE receiver context saved in
+{{request-parsing}}, `rctxt`.
+
+The output is a `response` to be sent to a Client.
+
+1. Let `cbor payload` equal the
+   [deterministically encoded CBOR](https://www.rfc-editor.org/rfc/rfc8949.html#name-deterministically-encoded-c)
+   `payload`. Return an empty `response` on CBOR encoding failure.
+1. Let `compressed payload` equal the [GZIP] compressed `cbor payload`, returning an empty
+   `response` on compression failure.
+1. Create a framed payload, as described in {{framing}}:
+    1. Create a `framing header`.
+    1. Set the `framing header` `Compression` to one of 2.
+    1. Set the `framing header` `Size` to the size of `compressed payload`.
+    1. Let `framed payload` equal the result of prepending the framing header to
+       `compressed payload`.
+    2. Padding MAY be added to `framing header`.
+    3. Return an empty `response` on failure of any of the previous steps.
+2. Let `response` equal the result of the encryption and encapsulation of `framed payload` with
+   `rctxt`, as described in {{response-encryption}}. Return an empty `response` on failure.
+3. Return `response`.
+
+### Parsing a Response
+
+TODO
+
+# IANA Considerations
```

Comment:
Add Security Considerations section.

Also IANA Considerations will likely include the new HTTP media types to be used with HTTP "content-type" headers (if specified in this spec).

Reply:
Done. Will start a doc on that.